src/Rule/Assertion/Declaration/ShouldNotExist/ShouldNotExistRule.php

```diff
@@ -0,0 +1,27 @@
+<?php declare(strict_types=1);
+
+namespace PHPat\Rule\Assertion\Declaration\ShouldNotExist;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\InClassNode;
+use PHPStan\Rules\Rule;
+
+/**
+ * @implements Rule<InClassNode>
+ */
+final class ShouldNotExistRule extends ShouldNotExist implements Rule
```

Comment:
To keep name consistency with the rest of the assertions, this class should be called `ExistsRule`.
The logic is normally placed in an Extractor so it's reusable for other rules.

The assertion `ShouldNotExist` would be the one implementing `getMessage()`, and `ExistsExtractor` the one implementing `getNodeType()` and `meetsDeclaration()`. Then this class (ExistsRule) can just `use ExistsExtractor`.

It's a bit over-complicated for two reasons:
- Custom rules in PHPStan can only _visit_ one type of node, while some assertions need to visit several (like shouldNotDepend)
- I use the 'extractor' traits not to repeat code between rules (shouldDepend, shouldNotDepend, canOnlyDepend...)

It might not be the best system for simple assertions like this one :sweat_smile: but it's the one in place and I think it's better to keep the same pattern everywhere.

Some extra context: in this module, 'Rule' refers to a PHPStan custom rule implementation, and 'Assertion' is the concept with meaning for PHPat tests. Conceptually, one assertion (ShouldNotDepend) is composed by one or more PHPStan rules (checking constant usage, checking class construction, checking docblocks...)